tests/mapped_struct.py

```diff
@@ -591,3 +593,43 @@ class MappedStringMappingBigTest(MappedStringMappingTest):
 class MappedString32MappingBigTest(MappedString32MappingTest):
     TEST_KEYS = [ "%d%s" % (i,k) for k in MappedStringMappingTest.TEST_KEYS for i in xrange(64) ]
     TEST_VALUES = [ v for v in MappedStringMappingTest.TEST_VALUES for i in xrange(64) ]
+
+class BsearchTest(unittest.TestCase):
+    SUPPORTED_DTYPES = [ numpy.uint32, numpy.int32, numpy.uint64, numpy.int64, 
+        numpy.double, numpy.single, numpy.float64, numpy.float32 ]
+
+    UNSUPPORTED_DTYPES = [ numpy.uint16, numpy.int16, numpy.uint8, numpy.int8 ]
+
+    for dtype in SUPPORTED_DTYPES:
+        def testBsearch(self, dtype=dtype):
+            testarray = range(1,101)
+            random.shuffle(testarray)
+            a = numpy.array(testarray[:50], dtype)
+            b = numpy.array([0] + testarray[50:] + range(101,103), dtype)
+            a = numpy.sort(a)
+            self.assertEqual(mapped_struct.bsearch(a, 0), 0)
+            self.assertEqual(mapped_struct.bsearch(a, 101), len(a))
+            self.assertEqual(mapped_struct.bsearch(a, 102), len(a))
+            for x in a:
+                ix = mapped_struct.bsearch(a, x)
+                self.assertLess(ix, len(a))
+                self.assertEqual(a[ix], x)
```

Comment:
Misising test for the case where the value is more than one time on the array: [1, 2, 2, 3, 4, 5], and that the first index should be returned on that case